

# Paso 1: Definir las hipótesis:

Hipótesis nula (H0): Las medias de las edades de los dos grupos son iguales.

Hipótesis alternativa (H1): Las medias de las edades de los dos grupos son diferentes.



# Paso 2: Recolectar y preparar los datos:

Vamos a generar dos conjuntos de datos ficticios para ilustrar el ejemplo.

In [1]:
import numpy as np

# Datos del Grupo 1 (por ejemplo, tiempo de respuesta)
grupo1 = np.array([23, 25, 27, 30, 32, 29, 28, 34, 26, 31])

# Datos del Grupo 2 (por ejemplo, tiempo de respuesta en un nuevo método)
grupo2 = np.array([32, 36, 40, 35, 38, 37, 33, 31, 29, 39])


In [2]:
grupo1.mean()

28.5

In [3]:
grupo2.mean()

35.0

#Paso 3: Realizar la prueba t de Student:

Usaremos la función ttest_ind de la biblioteca scipy.stats para realizar la prueba t de dos muestras independientes.

In [ ]:
import scipy.stats as stats

# Realizar la prueba t de dos muestras independientes
t_statistic, p_value = stats.ttest_ind(grupo1, grupo2)

# Mostrar los resultados
print("Estadística t:", t_statistic)
print("Valor p:", p_value)


Estadística t: -4.133991732024803
Valor p: 0.0006232691079705466


#Paso 4: Interpretar los resultados:

La estadística t es el valor calculado de la prueba t.
El valor p es la probabilidad de obtener un resultado tan extremo como el observado si la hipótesis nula es cierta.


Hipótesis Nula (H0): Las medias de los dos grupos son iguales.
Hipótesis Alternativa (H1): Las medias de los dos grupos son diferentes.

Asumimos un nivel de significancia (alfa) de 0.05 para esta prueba.

#Interpretación:

Estadística t: La estadística t es -4.133991732024803. Este valor representa cuántas desviaciones estándar está la diferencia entre las medias de los dos grupos. En este caso, es negativo, lo que indica que la media del Grupo 1 es menor que la del Grupo 2.

Valor p: El valor p es 0.0006232691079705466. Este valor es la probabilidad de obtener un resultado tan extremo como el observado si la hipótesis nula (H0) fuera cierta. En otras palabras, es la probabilidad de que las diferencias entre las medias de los grupos sean debidas al azar.

#Paso 5: Tomar una decisión:

Para tomar una decisión, compara el valor p con un nivel de significancia (alfa) predefinido. Por ejemplo, si alfa es 0.05 (nivel de confianza del 95%), puedes tomar la decisión de la siguiente manera:

Si p <= alfa, rechaza la hipótesis nula (H0).
Si p > alfa, no puedes rechazar la hipótesis nula (H0).
En este ejemplo, supongamos que alfa es 0.05. Luego, evaluamos el valor p:

In [ ]:
alfa = 0.05

if p_value <= alfa:
    print("Rechazamos la hipótesis nula (H0). Las medias son diferentes.")
else:
    print("No podemos rechazar la hipótesis nula (H0). Las medias son iguales.")

Rechazamos la hipótesis nula (H0). Las medias son diferentes.


#Decisión:

Dado que nuestro nivel de significancia (alfa) es 0.05, comparamos el valor p con alfa:

Si el valor p es menor o igual a alfa (0.05 en este caso), rechazamos la hipótesis nula (H0).
Si el valor p es mayor que alfa, no podemos rechazar la hipótesis nula (H0).
En este caso, el valor p (0.0006232691079705466) es mucho menor que alfa (0.05). Por lo tanto, podemos concluir que tenemos suficiente evidencia para rechazar la hipótesis nula (H0). Esto significa que las medias de los dos grupos son estadísticamente diferentes.

#Conclusión:

En resumen, los resultados de la prueba de hipótesis t de Student indican que las medias de los dos grupos son significativamente diferentes. Esto es consistente con nuestra hipótesis alternativa (H1).